make check: Failed: (test multi-hop remote force close on-chain htlc timeout)

### Background

```
$ make check
...
    --- PASS: TestLightningNetworkDaemon/test_multi-hop_local_force_close_on-chain_htlc_timeout (6.42s)
    --- FAIL: TestLightningNetworkDaemon/test_multi-hop_remote_force_close_on-chain_htlc_timeout (15.88s)
    	lnd_test.go:74: Failed: (test multi-hop remote force close on-chain htlc timeout): exited with error:
    		*errors.errorString unable to find bob's funding output sweep tx: wanted 1, found 0 txs in mempool: []
    		/home/maurycy/go/src/github.com/lightningnetwork/lnd/lnd_test.go:8141 (0xb9600a)
    			testMultiHopRemoteForceCloseOnChainHtlcTimeout: t.Fatalf("unable to find bob's funding output sweep tx: %v", err)
    		/home/maurycy/go/src/github.com/lightningnetwork/lnd/lnd_test.go:99 (0xb641fb)
    			(*harnessTest).RunTestCase: testCase.test(net, h)
    		/home/maurycy/go/src/github.com/lightningnetwork/lnd/lnd_test.go:10887 (0xbb3979)
    			TestLightningNetworkDaemon.func3: ht.RunTestCase(testCase, lndHarness)
    		/usr/lib/go/src/testing/testing.go:777 (0x4ef320)
    			tRunner: fn(t)
    		/usr/lib/go/src/runtime/asm_amd64.s:2361 (0x45c031)
    			goexit: BYTE	$0x90	// NOP
FAIL
exit status 1
FAIL	github.com/lightningnetwork/lnd	275.250s
make: *** [Makefile:147: itest] Error 1
```

### Your environment

* `lnd version 0.4.2-beta commit=8cd6eebadceef166d1b7ffb70df804ed0da9c81e`
* `Linux one 4.17.9-1-ARCH #1 SMP PREEMPT Sun Jul 22 20:23:36 UTC 2018 x86_64 GNU/Linux`
* `btcd version 0.12.0-beta`
* `Bitcoin Core Daemon version v0.16.1.0-gdac5d68fc6cf`
*  `go version go1.10.3 linux/amd64`

### Steps to reproduce

`make check`

Please see the attached `typescript` log.

### Expected behaviour

`PASS`

### Actual behaviour

`FAIL`

--- a/lnd_test.go
+++ b/lnd_test.go
@@ -220,24 +220,22 @@ func closeChannelAndAssert(ctx context.Context, t *harnessTest,
 	}
 	chanPointStr := fmt.Sprintf("%v:%v", txid, fundingChanPoint.OutputIndex)
 
-	// If we didn't force close the transaction, at this point, the channel
-	// should now be marked as being in the state of "waiting close".
-	if !force {
-		pendingChansRequest := &lnrpc.PendingChannelsRequest{}
-		pendingChanResp, err := node.PendingChannels(ctx, pendingChansRequest)
-		if err != nil {
-			t.Fatalf("unable to query for pending channels: %v", err)
+	// At this point, the channel should now be marked as being in the
+	// state of "waiting close".
+	pendingChansRequest := &lnrpc.PendingChannelsRequest{}
+	pendingChanResp, err := node.PendingChannels(ctx, pendingChansRequest)
+	if err != nil {
+		t.Fatalf("unable to query for pending channels: %v", err)
+	}
+	var found bool
+	for _, pendingClose := range pendingChanResp.WaitingCloseChannels {
+		if pendingClose.Channel.ChannelPoint == chanPointStr {
+			found = true
+			break
 		}
-		var found bool
-		for _, pendingClose := range pendingChanResp.WaitingCloseChannels {
-			if pendingClose.Channel.ChannelPoint == chanPointStr {
-				found = true
-				break
-			}
-		}
-		if !found {
-			t.Fatalf("channel not marked as waiting close")
-		}
+	}
+	if !found {
+		t.Fatalf("channel not marked as waiting close")
 	}
 
 	// We'll now, generate a single block, wait for the final close status
@@ -254,28 +252,26 @@ func closeChannelAndAssert(ctx context.Context, t *harnessTest,
 
 	// Finally, the transaction should no longer be in the waiting close
 	// state as we've just mined a block that should include the closing
-	// transaction. This only applies for co-op close channels though.
-	if !force {
-		err = lntest.WaitPredicate(func() bool {
-			pendingChansRequest := &lnrpc.PendingChannelsRequest{}
-			pendingChanResp, err := node.PendingChannels(
-				ctx, pendingChansRequest,
-			)
-			if err != nil {
+	// transaction.
+	err = lntest.WaitPredicate(func() bool {
+		pendingChansRequest := &lnrpc.PendingChannelsRequest{}
+		pendingChanResp, err := node.PendingChannels(
+			ctx, pendingChansRequest,
+		)
+		if err != nil {
+			return false
+		}
+
+		for _, pendingClose := range pendingChanResp.WaitingCloseChannels {
+			if pendingClose.Channel.ChannelPoint == chanPointStr {
 				return false
 			}
-
-			for _, pendingClose := range pendingChanResp.WaitingCloseChannels {
-				if pendingClose.Channel.ChannelPoint == chanPointStr {
-					return false
-				}
-			}
-
-			return true
-		}, time.Second*15)
-		if err != nil {
-			t.Fatalf("closing transaction not marked as fully closed")
 		}
+
+		return true
+	}, time.Second*15)
+	if err != nil {
+		t.Fatalf("closing transaction not marked as fully closed")
 	}
 
 	return closingTxid

--- a/lnd_test.go
+++ b/lnd_test.go
@@ -8130,21 +8130,17 @@ func testMultiHopRemoteForceCloseOnChainHtlcTimeout(net *lntest.NetworkHarness,
 		t.Fatalf(predErr.Error())
 	}
 
-	// We'll mine csvDelay blocks in order to generate the sweep transaction
-	// of Bob's funding output.
-	if _, err := net.Miner.Node.Generate(csvDelay - 1); err != nil {
-		t.Fatalf("unable to generate blocks: %v", err)
-	}
-
-	_, err = waitForTxInMempool(net.Miner.Node, 10*time.Second)
+	// Bob can sweep his output immediately.
+	_, err = waitForTxInMempool(net.Miner.Node, 20*time.Second)
 	if err != nil {
-		t.Fatalf("unable to find bob's funding output sweep tx: %v", err)
+		t.Fatalf("unable to find bob's funding output sweep tx: %v",
+			err)
 	}
 
 	// Next, we'll mine enough blocks for the HTLC to expire. At this
 	// point, Bob should hand off the output to his internal utxo nursery,
 	// which will broadcast a sweep transaction.
-	if _, err := net.Miner.Node.Generate(finalCltvDelta - csvDelay - 1); err != nil {
+	if _, err := net.Miner.Node.Generate(finalCltvDelta - 1); err != nil {
 		t.Fatalf("unable to generate blocks: %v", err)
 	}
 

